iothub/device/src/IotHubClientOptions.cs

```diff
@@ -6,14 +6,14 @@
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// Options that allow configuration of the device or module client instance during initialization.
+    /// Options that allow configuration of the IoT hub device or module client instance during initialization.
     /// </summary>
-    public class ClientOptions
+    public class IotHubClientOptions
```

Comment:
Named `IotHubClientOptions`, not IotHub**Device**ClientOptions because it is used both by device and module clients.

One option is to name it `IotHubDeviceClientOptions` and then subclass it just for naming purposes as IotHubModuleClientOptions. Thoughts?

Reply:
There shouldn't be any options that are exclusive to devices vs modules, so I'd prefer to just have one options class ```IotHubClientOptions```

Reply:
We have `IotHubServiceClientOptions` for the service client, so I am worried just `IotHubClientOptions` sounds like the superclass of it. As keeping a single client option for both device and module client is ideal, maybe we can have a new name for it? With that said, sorry I cannot come up with any better name for now.

Reply:
They will be in different namespaces and the doc comments should be clear it is for device clients.

FWIW, I don't mind calling it `IotHubDeviceClientOptions` anyway, and leave the ambiguity with the IotHubModuleClient which takes one as a parameter.